internal/k8s/gatewayclient/gatewayclient.go

```diff
@@ -545,22 +545,7 @@ func (g *gatewayClient) GetReferenceGrantsInNamespace(ctx context.Context, names
 		return nil, NewK8sError(err)
 	}
 	for _, refPolicy := range refPolicyList.Items {
-		refGrant := gwv1alpha2.ReferenceGrant{}
-		for _, refPolicyFrom := range refPolicy.Spec.From {
-			refGrant.Spec.From = append(refGrant.Spec.From, gwv1alpha2.ReferenceGrantFrom{
-				Group:     refPolicyFrom.Group,
-				Kind:      refPolicyFrom.Kind,
-				Namespace: refPolicyFrom.Namespace,
-			})
-		}
-		for _, refPolicyTo := range refPolicy.Spec.To {
-			refGrant.Spec.To = append(refGrant.Spec.To, gwv1alpha2.ReferenceGrantTo{
-				Group: refPolicyTo.Group,
-				Kind:  refPolicyTo.Kind,
-				Name:  refPolicyTo.Name,
-			})
-		}
-		refGrants = append(refGrants, refGrant)
+		refGrants = append(refGrants, gwv1alpha2.ReferenceGrant{Spec: refPolicy.Spec})
```

Comment:
e2e test for ReferencePolicy backwards compatibility fails as expected when this line is commented out, so the test coverage gap that existed when https://github.com/hashicorp/consul-api-gateway/pull/280 was discovered is covered now.